### PR TITLE
[various] Introduce explicit HttpClient "Accept" parameter

### PR DESCRIPTION
**Pull request by nupplaphil**
Part of https://github.com/friendica/friendica/pull/11389

### CODE REVIEW

**Review by MrPetovan — APPROVED** [review]

File: dwpost/dwpost.php [hunk 1]
```diff
@@ -192,7 +193,7 @@ function dwpost_send(App $a, array &$b)
 		Logger::debug('dwpost: data: ' . $xml);
 
 		if ($dw_blog !== 'test') {
-			$x = DI::httpClient()->post($dw_blog, $xml, ['Content-Type' => 'text/xml'])->getBody();
+			$x = DI::httpClient()->post($dw_blog, $xml, HttpClientAccept::DEFAULT, ['Content-Type' => 'text/xml'])->getBody();
```
Comment (annando) [inline dwpost/dwpost.php]: With a `post` we are transmitting the `content-type`, we aren't accepting stuff.

File: ijpost/ijpost.php [hunk 2]
```diff
@@ -186,7 +187,7 @@ function ijpost_send(&$a, &$b)
 		Logger::debug('ijpost: data: ' . $xml);
 
 		if ($ij_blog !== 'test') {
-			$x = DI::httpClient()->post($ij_blog, $xml, ['Content-Type' => 'text/xml'])->getBody();
+			$x = DI::httpClient()->post($ij_blog, $xml, HttpClientAccept::DEFAULT, ['Content-Type' => 'text/xml'])->getBody();
```
Comment (annando) [inline ijpost/ijpost.php]: See above.

File: ljpost/ljpost.php [hunk 3]
```diff
@@ -206,7 +207,7 @@ function ljpost_send(&$a,&$b) {
 		Logger::debug('ljpost: data: ' . $xml);
 
 		if ($lj_blog !== 'test') {
-			$x = DI::httpClient()->post($lj_blog, $xml, ['Content-Type' => 'text/xml'])->getBody();
+			$x = DI::httpClient()->post($lj_blog, $xml, HttpClientAccept::DEFAULT, ['Content-Type' => 'text/xml'])->getBody();
```
Comment (annando) [inline ljpost/ljpost.php]: Same :-)

File: webdav_storage/src/WebDav.php [hunk 4]
```diff
@@ -133,7 +134,7 @@ protected function hasItems(string $uri): bool
 	 */
 	protected function mkcol(string $uri): bool
 	{
-		return $this->client->request('mkcol', $uri, [HttpClientOptions::AUTH => $this->authOptions])
+		return $this->client->request('mkcol', $uri, HttpClientAccept::DEFAULT, [HttpClientOptions::AUTH => $this->authOptions])
```
Comment (annando) [inline webdav_storage/src/WebDav.php]: AFAIK the `mkcol` method isn't requesting data, so we shouldn't use that parameter there.

File: webdav_storage/src/WebDav.php [hunk 5]
```diff
@@ -177,7 +178,7 @@ protected function checkAndDeletePath(string $fullPath): void
 		foreach ($pathParts as $pathPart) {
 			$checkUrl = $this->url . $partURL;
 			if (!empty($partURL) && !$this->hasItems($checkUrl)) {
-				$response = $this->client->request('delete', $checkUrl, [HttpClientOptions::AUTH => $this->authOptions]);
+				$response = $this->client->request('delete', $checkUrl, HttpClientAccept::DEFAULT, [HttpClientOptions::AUTH => $this->authOptions]);
```
Comment (annando) [inline webdav_storage/src/WebDav.php]: Same as above.

File: webdav_storage/src/WebDav.php [hunk 6]
```diff
@@ -229,7 +230,7 @@ public function put(string $data, string $reference = ""): string
 			HttpClientOptions::AUTH => $this->authOptions,
 		];
 
-		$this->client->request('put', $this->url . '/' . $file[0], $opts);
+		$this->client->request('put', $this->url . '/' . $file[0], HttpClientAccept::DEFAULT, $opts);
```
Comment (annando) [inline webdav_storage/src/WebDav.php]: Same :-)

**Review by annando — CHANGES_REQUESTED** [review]
We should only use the `accept` header on `get` requests, since it isn't used for stuff like `post` and the other methods.

File: webdav_storage/src/WebDav.php [hunk 7]
```diff
@@ -6,6 +6,7 @@
 use Friendica\Core\Storage\Capability\ICanWriteToStorage;
 use Friendica\Core\Storage\Exception\ReferenceStorageException;
 use Friendica\Core\Storage\Exception\StorageException;
+use Friendica\Network\HTTPClient\Client\HttpClientAccept;
```
Comment (annando) [inline webdav_storage/src/WebDav.php]: Is this line still needed?

**Review by MrPetovan — APPROVED** [review]

**Pull request merged**